doc/tut3.md

```diff
@@ -253,6 +253,30 @@ The call to `myMacro` will generate the following code:
   echo MyType(a: 123.456'f64, b: "abcdef")
   ```
 
+You may also use `ident` in cases in which you wish to create an identifier from
+a string. For example you may create procedures which are named after a variable
+in the following way:
+
+  ```nim
+  import std/macros
+
+  macro createProcedures() =
+    result = newStmtList()
+
+    for i in 0..<10:
+      let name = ident("myProc" & $i)
+      let content = newLit("I am procedure number #" & $i)
+
+      result.add quote do:
```

Comment:
I think this can be more elegant using [newProc](https://nim-lang.org/docs/macros.html#newProc%2CopenArray%5BNimNode%5D%2CNimNode%2CNimNode) instead of ``quote do`` ?


``nim
  import std/macros
  macro createProcedures() =
    result = newStmtList()

    for i in 0..<10:
      let name = ident("myProc" & $i)
      let body = newStmtList()
      body.add newCall(ident("echo"), newStrLitNode("I am procedure number #" & $i))
      result.add newProc(name=name, body=body)
    
    debugEcho result.repr

  createProcedures()
  myProc7()
  ```

Reply:
I disagree, being able to use the normal Nim syntax is what's attractive and intuitive, in my opinion, rather than building the tree yourself and having to look up each node and its associated function.

In fact that was my process while trying to build this, I started with a template, then progressively switched to a macro but ideally I'd have liked not to have had to deal with the AST at all.

Reply:
I don't know what's the consensus is (or what @Araq think about it), but I've always found ``quote do`` less hygienic in macros.